src/index.ts

```diff
@@ -24,33 +23,41 @@ type GetUser = {
   email: string;
 };
 
+type $TSFixMe = any;
+
 export class AuthLiteClient {
   private secretKey: string;
   private apiKey: string;
   private orgId?: string;
-  private signedKey: string;
+  // INFO: using `!` ensuring that signedKey will be assigned before it is used and to suppress the TypeScript error
+  private signedKey!: string;
 
   constructor(apiKey: string, secretKey: string, orgId?: string) {
     this.secretKey = secretKey;
     this.apiKey = apiKey;
     this.orgId = orgId;
-    this.signedKey = this.jwtEncode(secretKey, { api_key: this.apiKey });
+    this.jwtEncode(this.secretKey, { api_key: this.apiKey });
```

Comment:
I will look into this, and please share steps to reproduce this.

Reply:
Hey @sahilsuman933 I have done the following to get to root cause:
- Brute force on generating secret using `crypto` and `TextEncoder`
- using available methods, and setting up isuer, audience, signedAt, expTime ...etc
- Observing error messages and network logs
- Looked into `jose` code and docs carefully

Then after trying again and again, I am pretty sure we now require to check why backend is sending `401` and `500` and what is causing over the backend to throw un-auth sign/token error.

Because from my side i have used and followed the same signature as it was done previously, same protection header and (yup) only difference is I am not passing setExpirationTime to false which is not available with `jose`.

Thus, we require to look at backend logs/code to check why 401.

Reply:
I presume that you are using NextJS? I forgot to tackle all the errors due to which It was sending 500 and `Internal Server Error` as errors on the frontend side while you were seeing the 401 errors in the terminal. I fixed that part and pushed the changes to the repository. So, you can pull the latest update and then try it.